newinterface.ts

```diff
@@ -0,0 +1,137 @@
+// Primary entry points into this API: `browser()` and `feature()` factory
+// functions.
+
+export declare function browser(key: browserKey): Browser;
+// You'd use `browser()` to query for browsers, such as `browser("chrome")` or
+// `browser("safari")`.
+
+export declare function feature(id: string): Feature;
+// And `feature()` to query for specific features, as in
+// `feature("dotted.path.to.feature")`.
+
+// Before getting into the weeds, I should note that, ultimately, some methods
+// of the objects returned from the factory functions are going to provide
+// simplified feature support interpretations. They'll be subject to some
+// options, but thanks to the work eliminating `true` and `null` values from
+// BCD, we can be more restrictve in the API than we can be in the schema.
+declare type supported =
+  | hasSupport // Supported
+  | lacksSupport // Unsupported
+  | hasSupportPartially // Supported, but only satisfied by one or more statements with `partial_implementation` set to `true`
+  | undefined; // Release not represented in a given feature
+// See the end for stronger definitions of the above.
+
+// The first of the "smart" wrappers around BCD objects. Here, a browser object
+// contains some releases and queries for getting at them.
+declare class Browser {
+  id: browserKey;
+  data: object; // Underlying BCD browser object
+
+  // Pass through BCD values where convenient
+  name: string;
+  type: string;
+  preview_name: string;
+  // etc.
+
+  constructor(id: browserKey, data: unknown);
+
+  releaseAt(index: number): Release | undefined;
```

Comment:
I don't see a `length` property, so is the API to keep querying until undefined is returned? That works, but it's quite unusual.